app/src/main/java/com/rage/clamber/Fragments/HomeFragment.java

```diff
@@ -1,33 +1,117 @@
+import com.rage.clamber.Data.WallSection;
+import com.rage.clamber.Networking.ApiManager;
 import com.rage.clamber.R;
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.Bind;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 /**
  * Displays the home page.
  */
 public class HomeFragment extends Fragment {
 
+    public User mainUser;
+    public static final String TAG = HomeFragment.class.getSimpleName();
+    public List<WallSection> wallSections;
+    @Bind(R.id.home_page_image_1_grid_view)
+    ImageView wallImage1;
+    @Bind(R.id.home_page_image_2_grid_view)
+    ImageView wallImage2;
+    @Bind(R.id.home_page_image_3_grid_view)
+    ImageView wallImage3;
+
 
     public HomeFragment() {
         // Required empty public constructor
     }
 
-    public static HomeFragment newInstance() {
-        return new HomeFragment();
+    public static HomeFragment newInstance(User user) {
+        Bundle args = new Bundle();
+        args.putParcelable(HomePage.ARG_USER, user);
+        HomeFragment fragment = new HomeFragment();
+        fragment.setArguments(args);
+        return fragment;
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_home, container, false);
+
+        View rootView = inflater.inflate(R.layout.fragment_home, container, false);
+        ButterKnife.bind(this, rootView);
+        wallSections = new ArrayList<>();
+        mainUser = getArguments().getParcelable(HomePage.ARG_USER);
+        getLastUpdatedWallSections(mainUser.getUserName());
+
+        return rootView;
+    }
+
+    public void getLastUpdatedWallSections(String username){
+        ConnectivityManager connectivityManager = (ConnectivityManager) getActivity().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+        if (networkInfo != null && networkInfo.isConnected()){
+            final Call<List<WallSection>> lastUpdateWallCall = ApiManager.getClamberService().getLastUpdatedWalls(mainUser.getUserName());
+            lastUpdateWallCall.enqueue(new Callback<List<WallSection>>() {
+                @Override
+                public void onResponse(Call<List<WallSection>> call, Response<List<WallSection>> response) {
+                    if (response.code() == 200) {
+                        List<WallSection> lastUpdateWallSections = response.body();
+                        wallSections.addAll(lastUpdateWallSections);
+
+                        Picasso.with(getActivity()).load(ApiManager.getImageUrl("assets/wall_sections/" + wallSections.get(0).getId() + ".jpg")).fit().centerCrop().into(wallImage1);
```

Comment:
ok, there's probably some way you could do a loop, and go through the first up-to-3 elements of the returned list, but given that android sort of makes you hardcode which elements you're replacing, and you can guarantee there'll always be at least three elements returned, it's probably fine as-is. Just wanted to make sure you weren't going to get any index out of bounds exceptions.